File: src/main/java/ru/donz/wf/behaviors/FlowState.java
package ru.donz.wf.behaviors;

import lombok.Data;
import lombok.extern.slf4j.Slf4j;

@Data
@Slf4j
public class FlowState {
    private final String txId;
    private final String lastMessageType;
    private final String lastMessage;

    public FlowState newIncomingExternalMessage(FlowEvent.IncomingExternalMessage externalMessage) {
        log.info("Received incoming external message {}", externalMessage.getMessage());
        return new FlowState(txId, lastMessageType, externalMessage.getMessage());
    }
}

File: src/main/kotlin/ru/onix/wf/behaviors/FlowState.kt
package ru.onix.wf.behaviors

import ru.onix.wf.log

data class FlowState(
    val txId: String,
    val lastMessageType: String,
    val lastMessage: String,
) {
    fun newIncomingExternalMessage(externalMessage: FlowEvent.IncomingExternalMessage): FlowState {
        log().info("Received incoming external message {}", externalMessage.message);
        return FlowState(txId, lastMessageType, externalMessage.message);
    }
}